src/components/TheHeader.vue

```diff
@@ -0,0 +1,81 @@
+<script lang="ts" setup>
+import MoonIcon from "@/assets/icons/MoonIcon.vue";
+import { ECountriesMode } from "@/types/countries";
+import type { Ref } from "vue";
+import { computed, ref } from "vue";
+
+const themeMode: Ref<ECountriesMode> = ref(ECountriesMode.LIGHT_MODE);
+const themeModeTile = computed(() =>
+  themeMode.value === ECountriesMode.LIGHT_MODE ? "Light Mode" : "Dark Mode"
+);
+
+const changeTheme = () => {
+  themeMode.value === ECountriesMode.LIGHT_MODE
+    ? (themeMode.value = ECountriesMode.DARK_MODE)
+    : (themeMode.value = ECountriesMode.LIGHT_MODE);
+
+  document.querySelector(":root")?.classList.toggle("dark-theme");
+};
+</script>
+
+<template>
+  <header class="c-header">
+    <div class="c-header__elements-wrapper">
+      <RouterLink class="c-header__title" to="/"
+        >Where is the world?
+      </RouterLink>
+      <button class="c-header__theme-container" @click="changeTheme()">
+        <MoonIcon class="c-header__moon-icon" />
+        <span class="c-header__theme-mode-title">{{ themeModeTile }}</span>
+      </button>
+    </div>
+  </header>
+</template>
+
+<style lang="scss" scoped>
+.c-header {
+  height: 8.2rem;
+  background-color: var(--header-color);
+  box-shadow: var(--shadow);
+  display: flex;
+  align-items: center;
+
+  &__elements-wrapper {
```

Comment:
jeśli masz mieć takie same wartości dla headera i jeszcze jakiś innych elementów, to myślę, że warto zastanowić się nad wyniesieniem tego to jakiego utility class .container, gdzie będziesz miał max-width, margin-inline ustawione. Możesz też pomyśleć o zmiennych css w tej klasie, które będą sterować wartością max-width, jeśli będziesz potrzebował